pkg/custom-provider/provider.go

```diff
@@ -98,8 +98,20 @@ func (p *prometheusProvider) metricFor(value pmodel.SampleValue, name types.Name
 		Metric: custom_metrics.MetricIdentifier{
 			Name: info.Metric,
 		},
+		
 		// TODO(directxman12): use the right timestamp
-		Timestamp: metav1.Time{Time: time.Now()},
+		
+		
+		/*Making the changes in the timestamp as per issue
+		number #545*/
+
+		// Previously set timestamp
+		//Timestamp: metav1.Time{Time: time.Now()},
+
+		// Newly set timestamp
+		Timestamp: metav1.Time{
+			Time: value.Timestamp.Time(),
```

Comment:
This will not compile.
`value` is a [`github.com/prometheus/common/model.SampleValue`](https://pkg.go.dev/github.com/prometheus/common/model#SampleValue), which has no `Timestamp` field.
You need to change `metricFor` to take a [`github.com/prometheus/common/model.Sample`](https://pkg.go.dev/github.com/prometheus/common/model#Sample) or a struct which includes both `Value` and `Timestamp`.
To do that, you'll need to refactor every function which calls `metricFor`.

Reply:
@olivierlemasle I have refactored the metricFor function in the provider.go class to add the Sample model class.